src/plonk/prover.rs

```diff
@@ -529,50 +529,65 @@ impl<C: CurveAffine> Proof<C> {
                 })
                 .or_else(|| Some(poly));
         }
-        let mut f_poly = f_poly.unwrap();
-        let mut f_blind = Blind(C::Scalar::random());
 
-        let f_commitment = params.commit(&f_poly, f_blind).to_affine();
+        let f_poly = f_poly.unwrap();
+        let mut f_blind = Blind(C::Scalar::random());
+        let mut f_commitment = params.commit(&f_poly, f_blind).to_affine();
 
-        hash_point(&mut transcript, &f_commitment)?;
+        let final_q_evals;
 
-        let x_6: C::Scalar = get_challenge_scalar(Challenge(transcript.squeeze().get_lower_128()));
+        let opening = loop {
```

Comment:
nit: this could be `let (opening, q_evals) = loop {` and you could get rid of `final_q_evals`. :)